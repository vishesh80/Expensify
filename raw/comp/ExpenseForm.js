import React from 'react';

import moment from 'moment';
import {SingleDatePicker} from 'react-dates';

import 'react-dates/initialize';



class ExpenseForm extends React.Component
{
    constructor(props)
    {
        super();

        if(!props.initialState) this.state = {


            description: "",
            amount:  "",
            note: "",
            createdAt: moment(),

            focused: null

        };

        else this.state = {

            description: (props.initialState.description) ? props.initialState.description : "",
            amount: (props.initialState.amount) ? props.initialState.amount : "",
            note: (props.initialState.note) ? props.initialState.note : "",
            createdAt: (props.initialState.createdAt) ? props.initialState.createdAt : moment(),

            focused: null
        }

        this.onAchange = this.onAchange.bind(this);
        this.onNchange = this.onNchange.bind(this);
        this.onDchange = this.onDchange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    onDchange(e){
        e.persist();
        this.setState(prev => ({
            description: e.target.value,
        }));
    }

    onAchange(e){
        e.persist();
        if(!(/^\d*(\.\d{0,2})?$/.test(e.target.value))) return;
        this.setState(prev => ({
            amount: e.target.value,
        }));
    }

    onNchange(e){
        e.persist();
        this.setState(prev => ({
            note: e.target.value,
        }));
    }

    onSubmit(e){
        e.persist();

        
        if(this.state.description && this.state.amount)
        {
            console.log(this.state.createdAt);
            if(!(/^[a-z\sA-Z]+$/.test(this.state.description.trim())) || 
               !(this.state.createdAt)) return alert("Fill the valid information.");
            
                this.props.submit({
                description : this.state.description,
                amount : this.state.amount,
                note: this.state.note,
                createdAt: this.state.createdAt
            });               
        }
        else return alert("Please Complete the form.");
    }

    render()
    {
        return (<div>
        <input type="text" value={this.state.description} placeholder="Description" onChange={this.onDchange}/><br />
        <input type="text" value={this.state.amount} placeholder="Amount" onChange={this.onAchange}/><br />

        <textarea cols="30" rows="10" value={this.state.note} placeholder="Note (optional)" onChange={this.onNchange}></textarea><br/>
            
        <SingleDatePicker
        date={this.state.createdAt} // momentPropTypes.momentObj or null
        onDateChange={date => this.setState({createdAt: date})} 
        focused={this.state.focused} // PropTypes.bool
        onFocusChange={({ focused }) => this.setState({focused})} // PropTypes.func.isRequired
        numberOfMonths={1}
        isOutsideRange={d => false}
        id="your_unique_id" />
        <br/>

        <button onClick={this.onSubmit}>Submit</button>
        </div>);     
    }
}


export default ExpenseForm;