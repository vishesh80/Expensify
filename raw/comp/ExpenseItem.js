import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

import {remove} from '../modules/actions'


export const ExpenseItem = ({ description, amount, id, createdAt, dispatch}) => (
    <li>
        <h1>{description}</h1>
        <p>{amount + "$ - "+ createdAt.format("Do MMMM YYYY")}</p>
        <button onClick={e => dispatch(remove(id))}>Remove</button>
        <NavLink to={"/edit/"+id} exact={true}>Edit</NavLink>
    </li>
);

export default connect()(ExpenseItem);