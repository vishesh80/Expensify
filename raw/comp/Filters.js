import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { DateRangePicker } from 'react-dates';
import {setTextFilter , sortByDate as sd, sortByAmount as sa,setEndDate,setStartDate} from '../modules/actions';

class Filters extends React.Component{

    constructor(p)
    {
        super();
        this.state = {
            startDate: p.startDate,
            endDate: p.endDate,
            focused: null
        };
    }

    render()
    {
        return (
            <div>
                <label>Set Text Filter</label>
                <input type="text" value={this.props.text} onChange={e => this.props.dispatch(setTextFilter(e.target.value))} />
                <br />
                <br />
                <label>Sort By</label>
                <select value={this.props.sortBy} onChange={e => (e.target.value === "amount") ? this.props.dispatch(sa()) : this.props.dispatch(sd())}>
                    <option value="amount">Amount</option>
                    <option value="date">Date</option>
                </select>
                <br />
                <br />
                <DateRangePicker
                    startDate={this.state.startDate} // momentPropTypes.momentObj or null,
                    startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
                    endDate={this.state.endDate} // momentPropTypes.momentObj or null,
                    endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
                    onDatesChange={({ startDate, endDate }) => {
                        
                        this.setState({ startDate, endDate });
                        this.props.dispatch(setStartDate(startDate));
                        this.props.dispatch(setEndDate(endDate));
                    }} 
                    focusedInput={this.state.focused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={focused => this.setState({ focused })} // PropTypes.func.isRequired,
                    showClearDates={true}
                    isOutsideRange={() => false}
                    numberOfMonths={1}
                />
            </div>
        ); 
    }
}  

const mapStateToProps = state => ({
    text: state.filters.text,
    sortBy : state.filters.sortBy,
    startDate : state.filters.startDate,
    endDate : state.filters.endDate
}); 


export default connect(mapStateToProps)(Filters);