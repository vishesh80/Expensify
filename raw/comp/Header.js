import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (<div>
    <h1>Expense Tracker</h1>
    <NavLink to="/" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" exact={true}>Create Expense</NavLink>
    <NavLink to="/help" exact={true}>Help</NavLink>
</div>);

export default Header;