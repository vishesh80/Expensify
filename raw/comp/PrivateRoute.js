import React from 'react';
import { Route , Redirect} from 'react-router-dom';
import { connect } from 'react-redux';

const route = props => (
                        <Route {...props} 
                                component={(props.auth)? props.component : redirect} />
                        );


const mapStateToProps = state => ({

    auth : !!state.auth.uid
});

const redirect = () => (<Redirect to="/" />);

export default connect(mapStateToProps)(route);