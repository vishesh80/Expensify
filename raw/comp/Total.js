import React from 'react';
import filter from '../modules/filter';

const Total = props => {

    const array = filter(props.expenses, props.filters);
    const length = array.length;
    
    return (
    <div>
        <h3>Viewing {(length === 1)? length+" expense": length+" expenses"} totalling ${
            array
            .map(e => Number(e.amount))
            .reduce((sum,value) => sum + value,0)
        }</h3>
    </div>
    );
}

export default Total;
 

