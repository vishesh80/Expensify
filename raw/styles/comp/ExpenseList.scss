.expenses
{
    width: 80%;
    margin: auto;
    margin-top: 2em;
    padding-top: 1px;
    border: gainsboro solid 1px;
    list-style-type: none;
    padding: 0;

    
}

.expense
{
    display: flex;
    border-bottom: rgba(220, 220, 220, 0.534) solid  1px;
}

.left
{
    padding: .2em 1.5em;
    max-width: 75%;

    h1
    {
        font-size: 2rem;
        font-weight: 300;
        margin: .5em 0em;
        word-break: break-all;

    }
}

.right
{
    margin-left: auto;
    min-width: 106px;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    
    p
    {
        font-size: 2rem;
        padding-right: 1em;
    }
}



.remove,.edit
{
     
        background-color: rgb(32, 166, 255);
        font-size: 1rem;
        padding: .5em 1em;
        border: none;
        color: white;
        text-decoration: none;
        display: inline-block;
       
}
.remove
{
    background-color: rgb(255, 96, 96);
    cursor: pointer;
}

