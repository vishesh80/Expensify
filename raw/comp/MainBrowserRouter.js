import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import Header from './Header';
import d from './Pages/Dashboard';
import c from './Pages/CreateExpenses';
import h from './Pages/Help';
import e from './Pages/Edit';


const _404 = () => (<h1>404</h1>);

export default function mainBrowserRouter()
{
    return (
        <BrowserRouter>
            <Header />
            <Switch>
                <Route path="/" component={d} exact={true} />
                <Route path="/create" component={c} exact={true} />
                <Route path="/help" component={h} exact={true} />
                <Route path="/edit/:id" component={e} exact={true} />
                <Route component={_404} exact={true} />
            </Switch>
        </BrowserRouter>
    );
}