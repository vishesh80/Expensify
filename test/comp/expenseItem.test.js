import { shallow } from "enzyme";
import moment from 'moment';
import React from "react";
import { ExpenseItem } from "../../raw/comp/ExpenseItem";
import eToJ from "enzyme-to-json";

test("ExpenseItem component test", () => {

    const props = {
      description: "eg",
      amount: 12,
      createdAt: moment(0),
      id: "eg",
      dispatch: function (r) {
        return r;
      }
    };

    let wrapper = shallow(<ExpenseItem {...props} />);
    expect(eToJ(wrapper)).toMatchSnapshot();
});
