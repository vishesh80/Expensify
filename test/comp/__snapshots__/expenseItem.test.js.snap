// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseItem component test 1`] = `
<li>
  <h1>
    eg
  </h1>
  <p>
    12$ - 1st January 1970
  </p>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
  <[object Object]
    exact={true}
    to="/edit/eg"
  >
    Edit
  </[object Object]>
</li>
`;
